quandl/model/datatable.py

```diff
@@ -14,3 +29,57 @@ def get_path(cls):
     def data(self, **options):
         updated_options = Util.convert_options(**options)
         return Data.page(self, **updated_options)
+
+    def download_file(self, file_or_folder_path, **options):
+        if not isinstance(file_or_folder_path, str):
+            raise QuandlError(Message.ERROR_FOLDER_ISSUE)
+
+        if 'params' not in options:
+            options['params'] = {}
+
+        return self._url_request(file_or_folder_path, **options['params'])
+
+    def _url_request(self, file_or_folder_path, **options):
```

Comment:
could you refactoring this method smaller and easy to read?

Reply:
updated